### ISSUE TITLE
Choose format for server token

### ISSUE THREAD

**Issue by nissayeva** [backstage/backstage#7352]
As a first pass for backend to backend authorization, perhaps we can create a shared HMAC key for all backends that would be accessible from `app-config.yaml`. Where to store the key itself is still an open question.

**Comment by erikxiv** [comment]
Some thoughts:
* We could reuse the JWT format used by the identity token, possibly with different properties. Less logic and more easy to reason about. What data to put in it can be deferred until later.
* We could reuse JWK signatures. One of the current blockers is sharing the secret among backends (auth-plugin creates and store these locally, and with a short expiration date), but a manual process where adopters generate a key using https://mkjwk.org/ and adds it to `app-config.yaml` should easily work, allowing further reuse of code between client/server logic. Admittedly the secret is rather bulky compared to shorter HMAC secret, and there is no extra value gained in the asymmetric key for this server-to-server scenario.
* We could allow adopters to plugin a custom token, for those that already have an existing token infrastructure in place they want to reuse.

**Comment by freben** [comment]
Yeah the initial gut feeling is that we can start with a pre-shared static symmetric key, and then see where we need to evolve from there.

Sounds sensible to encode that as a JWK nested with a type so that we can expand on it. Something like this in `app-config`?

```yaml
catalog:
  authorization:
    type: jwk
    key: >
      {
        "kty": "oct",
        "alg": "dir",
        "k": "..."
      }
```

or of course using a `$include` from a file on disk.

The question is, should we start thinking about whether different clients should have different keys so we can easily revoke/rotate just one of them? Perhaps requiring a `"kid"` on them.

I may be out on thin ice here and @Rugvip will probably have more precise input :)

**Comment by erikxiv** [comment]
> The question is, should we start thinking about whether different clients should have different keys so we can easily revoke/rotate just one of them? Perhaps requiring a "kid" on them.

Sounds like an easy and valuable addition: config allows multiple keys, top one used to sign but the rest are accepted for validation. Also plays well with identity token validation that uses `kid` to find the appropriate key.

**Comment by nissayeva** [comment]
the first pass of this is implemented in #7943
